src/circuit/gadget/ecc/chip/mul_fixed/short.rs

```diff
@@ -0,0 +1,253 @@
+use std::array;
+
+use super::super::{copy, CellValue, EccConfig, EccPoint, EccScalarFixedShort, Var};
+use crate::constants::ValueCommitV;
+
+use halo2::{
+    circuit::Layouter,
+    plonk::{ConstraintSystem, Error, Selector},
+    poly::Rotation,
+};
+use pasta_curves::pallas;
+
+pub struct Config<const NUM_WINDOWS: usize> {
+    // Selector used for fixed-base scalar mul with short signed exponent.
+    q_mul_fixed_short: Selector,
+    super_config: super::Config<NUM_WINDOWS>,
+}
+
+impl<const NUM_WINDOWS: usize> From<&EccConfig> for Config<NUM_WINDOWS> {
+    fn from(config: &EccConfig) -> Self {
+        Self {
+            q_mul_fixed_short: config.q_mul_fixed_short,
+            super_config: config.into(),
+        }
+    }
+}
+
+impl<const NUM_WINDOWS: usize> Config<NUM_WINDOWS> {
+    // We reuse the constraints in the `mul_fixed` gate so exclude them here.
+    // Here, we add some new constraints specific to the short signed case.
+    pub(crate) fn create_gate(&self, meta: &mut ConstraintSystem<pallas::Base>) {
+        meta.create_gate("Short fixed-base mul gate", |meta| {
+            let q_mul_fixed_short = meta.query_selector(self.q_mul_fixed_short);
+            let y_p = meta.query_advice(self.super_config.y_p, Rotation::cur());
+            let y_a = meta.query_advice(self.super_config.add_config.y_qr, Rotation::cur());
+            let sign = meta.query_advice(self.super_config.window, Rotation::cur());
+
+            // `(x_a, y_a)` is the result of `[m]B`, where `m` is the magnitude.
+            // We conditionally negate this result using `y_p = y_a * s`, where `s` is the sign.
+
+            // Check that the final `y_p = y_a` or `y_p = -y_a`
+            let y_check = q_mul_fixed_short.clone()
+                * (y_p.clone() - y_a.clone())
+                * (y_p.clone() + y_a.clone());
+
+            // Check that the correct sign is witnessed s.t. sign * y_p = y_a
+            let negation_check = sign * y_p - y_a;
+
+            array::IntoIter::new([y_check, negation_check])
+                .map(move |poly| q_mul_fixed_short.clone() * poly)
+        });
+    }
+
+    pub fn assign(
+        &self,
+        mut layouter: impl Layouter<pallas::Base>,
+        scalar: &EccScalarFixedShort,
+        base: &ValueCommitV,
+    ) -> Result<EccPoint, Error> {
+        layouter.assign_region(
+            || "Short fixed-base mul",
+            |mut region| {
+                let offset = 0;
+
+                // Copy the scalar decomposition
+                self.super_config
+                    .copy_scalar(&mut region, offset, &scalar.into())?;
+
+                let (acc, mul_b) = self.super_config.assign_region_inner(
+                    &mut region,
+                    offset,
+                    &scalar.into(),
+                    base.clone().into(),
+                    self.super_config.q_mul_fixed,
+                )?;
```

Comment:
I suspect also that making the 6 columns be on the right would help with layout, since we have a lot of regions that use columns on the left. But we can tweak things like that after we've merged the Action circuit PR.